lib/hosting/leaseweb_apis.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+require "excon"
+class Hosting::LeasewebApis
+  def initialize(leaseweb_host)
+    @host = leaseweb_host
+  end
+
+  def get_main_ip4
+    connection = Excon.new(@host.connection_string,
```

Comment:
We can do something similar there
https://github.com/ubicloud/ubicloud/pull/2563